### DISCUSSION TITLE
Définition d'une licence

### DISCUSSION THREAD

**Discussion by JulienWilhelm** [cnumr/EcoIndex#159] [General]
Pour faire écho au mail de Frédéric,

Nous devons fixer une licence afin de valoriser et préserver le travail accompli pour EcoIndex.

La licence suivante a été évoquée : [CC BY-NC-ND](https://creativecommons.org/licenses/by-nc-nd/2.0/fr/). Celle-ci est assez restrictive en cela qu'elle interdit toute commercialisation de l'outil et modifications partagées, mais c'est a priori l'ambition.

Qu'en dites-vous ?
Et aussi, où commence et où se termine cette licence ? 
Quid des licences éventuelles applicables au coeur du projet source ?

**Comment by vvatelot** [comment]
Hello, j'ai fait une PR sur le repo du site ecoindex : https://github.com/cnumr/EcoIndex/pull/157

Je pense qu'il faudrait basculer les licences des différents projets du collectif sur cette même licence et ajouter un paragraphe type expliquant ça...

**Reply by vvatelot** [reply]
Par contre, je pense à ça, mais peut être qu'il faut également l'exposer directement sur le site et pas uniquement sur le repo du projet... :thinking:

**Reply by JulienWilhelm** [reply]
Assez d'accord avec toi pour le côté "uniformisation". 
Quant au côté communication, il me parait tout aussi essentiel.

Sait-on si on avait une ou plusieurs licenses définies initialement sur Ecoindex ?
Parce qu'il y a, à minima : 
- L'algorithme (ou la méthode)
- Le service

**Reply by vvatelot** [reply]
Pour info, le texte explicatif donnerait quelque chose comme ça: https://github.com/cnumr/EcoIndex/tree/doc/licence#licence (en FR) et https://github.com/cnumr/EcoIndex/tree/doc/licence#licence (EN)

J'ai piqué le texte sur https://github.com/cnumr/best-practices-wordpress que je trouvais plutôt pas mal fait.

Qu'entends tu par "le service" ?

**Reply by JulienWilhelm** [reply]
Oui, pour l'explication, pas besoin de faire compliqué (par contre, privilégions plutôt les URL en https).

Le service, c'est le site web en lui-même (ou wepApp selon le point de vue) : contenu, interprétation des résultats, etc.

**Comment by bordage** [comment]
Bonjour,

On ne peut pas protéger une algorithme. 

En revanche, on peut protéger :
- le service EcoIndex.fr et les résultats qu'il délivre avec une licence CC-By-NC-ND ;
- les facteurs d'impacts à partir desquels on calcule l'empreinte, qui sont ma propriété.

Il n'y a que la licence CC-By-NC-ND qui ait été adossée à EcoIndex.fr jusqu'à présent.

J'ai choisi cette licence pour les raisons suivantes : 
- By : pour le respect de la paternité 
- NC : pour éviter que des petits malins qui ne contribuent pas se fasse de l'argent sur notre dos. 
- ND : pour concentrer toute l'énergie sur un seul projet en évitant les forks.
 
Malgré la licence et le fait que les facteurs d'impacts sont ma propriété que personne ne peut exploiter sans mon consentement (ce qui était censé protéger un peu notre travail du pillage), j'ai compté 20+ services concurrents à EcoIndex.fr ces 2 derniers années. Pratiquement aucun ne respectent la licence... 

Voici le texte que nous avons trouvé avec Lise pour compléter la licence CC : 
"The LCA values used by **tool_name** to evaluate environmental impacts are not under free license - © Frédéric Bordage
Please also refer to the mentions provided in the code files for specifics on the IP regime."
Exemple : https://gitlab.com/fabernovel/heart/-/blob/2bbb77578ed045ea15b7a51fadf2a0abca7ce657/modules/heart-greenit/README.md

**Comment by vvatelot** [comment]
OK, de mon côté, j'ai fait les PR pour utiliser la licence CC-By-NC-ND et le texte à ajouter dans le readme : 
https://github.com/cnumr/ecoindex_api/pull/206
https://github.com/cnumr/ecoindex_python/pull/94
https://github.com/cnumr/ecoindex_scrap_python/pull/44
https://github.com/cnumr/ecoindex_python/pull/94

Si OK pour vous, je valide

**Comment by bordage** [comment]
OK pour moi.

Attention à bien associer en plus une licence adaptée pour protéger ton
code source .

Fred

+33 6 16 95 96 01


Le ven. 4 nov. 2022 à 13:40, Vincent Vatelot ***@***.***> a
écrit :

**Comment by JulienWilhelm** [comment]
Idem, OK de mon côté.
Ne faudrait-il pas attendre l'avis de plus de monde ?
